stage1_fly/enter/main.go

```diff
@@ -47,10 +50,30 @@ func getRootDir(pid string) (string, error) {
 	return os.Readlink(rootLink)
 }
 
-func execArgs() error {
+// readEnv reads the environment from the env file written by stage1 run
+func readEnv() ([]string, error) {
```

Comment:
I would prefer this to receive the base directory from the outside, instead of performing a `Getwd()` internally.